pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -462,6 +467,74 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileBuildRun) updateBuildRunUsingTaskRunCondition(ctx context.Context, buildRun *buildv1alpha1.BuildRun, taskRun *v1beta1.TaskRun, trCondition *apis.Condition) error {
+	var reason, message string = trCondition.Reason, trCondition.Message
+
+	switch v1beta1.TaskRunReason(reason) {
+	case v1beta1.TaskRunReasonTimedOut:
+		reason = "BuildRunTimeout"
+		message = fmt.Sprintf("BuildRun %s failed to finish within %s",
+			taskRun.GetLabels()[buildv1alpha1.LabelBuildRun],
+			taskRun.Spec.Timeout.Duration,
+		)
+
+	case v1beta1.TaskRunReasonFailed:
+		if taskRun.Status.CompletionTime != nil {
```

Comment:
What happens if the TaskRun is not complete? Would we expect this to be a time out error?

Reply:
In theory no. If the TaskRun Condition.Reason is `Failed`, this means that something failed in the Pod and Completion will be set always after the `Failed` is set in the TaskRun Condition.Reason. You can dive in the [code](https://github.com/tektoncd/pipeline/blob/66096db2c12125693c39b32bb1cbfeff6cfdb644/pkg/pod/status.go#L272-L316) to verify that part.

Also, Tekton have this nice table that illustrates the above, see [table](https://github.com/tektoncd/pipeline/blob/master/docs/taskruns.md#monitoring-execution-status) from tekton. You will find there that whevener the `Reason` is `Failed`, the CompletionTime is set.

Therefore, to answer your question, a `TaskRun` cannot be `Failed` and "not complete" .